### PR TITLE
google-cloud-sdk: bump python dependency version from 3.9 to 3.10

### PR DESCRIPTION
**Pull request by jrabinow**
#### Description

google-cloud-sdk: bump python from 3.9 to 3.10

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 12.2 21D49 x86_64
Xcode 13.1 13A1030d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by jrabinow** [comment]
My bad, I hadn't seen that. Thank you for flagging this issue. It looks like this package already has python3.9 declared as dependency so we're already in incompatible territory today.

As far as I can tell, the `gcloud` command completely ignores the dependency version in the portfile and invokes `/opt/local/bin/python3` directly, meaning that we could put python3.12 in the portfile and the `gcloud` command would still work as long as there's a python binary (or a symlink to the python binary) at the correct location on the filesystem.

Except that we have a bug [here](https://issuetracker.google.com/issues/205005959) which I ran into yesterday. I had switched the python version back to 3.9 in the port file and noticed the bug was still there, so I thought it didn't make a difference. It turns out that sticking with python 3.9 DOES avoid the bug, but the version is not set in the portfile definition, it's set using `port select --set python3 python39`

Basically, the python version in the portfile doesn't make a difference as far as I can tell, but I still think the best course of action here is to wait until Google fixes the bug before bumping python versions. We can close this PR in the meantime. What do you think?

**Comment by ryandesign** [comment]
> this package already has python3.9 declared as dependency so we're already in incompatible territory today

According to https://issuetracker.google.com/issues/170125513 it's working fine with python 3.9 as of version 319.0.0.

> `gcloud` command completely ignores the dependency version in the portfile and invokes `/opt/local/bin/python3` directly

The port must be fixed so that it runs the python that's declared as a dependency in the portfile.

> we have a bug [here](https://issuetracker.google.com/issues/205005959)

That does seem to say that it's incompatible with python 3.10. As such, this PR should be closed until that's resolved.

**Comment by jrabinow** [comment]
@ryandesign:

> The port must be fixed so that it runs the python that's declared as a dependency in the portfile.

That may be an interesting task, it looks like the `gcloud` command is a shellscript wrapper that attempts to find all python versions installed on the system and uses the one it likes best (not quite true but close enough) to launch the main python codebase. What you're suggesting may be doable with a patch, and I understand the want for actually using the declared dependencies and nothing else, but I also suspect this is going to turn into an uphill battle against Google, potentially at every release.

> > we have a bug [here](https://issuetracker.google.com/issues/205005959)
> 
> That does seem to say that it's incompatible with python 3.10. As such, this PR should be closed until that's resolved.

Will do, @breun I will let you close this to make sure you get the chance to share your thoughts on the subject.
